ConnectorCSI/ConnectorCSIShared/UI/ConnectorBindingsCSI.Recieve.cs

```diff
@@ -144,68 +145,68 @@ private List<ApplicationObject> ConvertReceivedObjects(ISpeckleConverter convert
     }
 
     /// <summary>
-    /// Recurses through the commit object and flattens it.
+    /// Traverses the object graph, returning objects to be converted.
     /// </summary>
-    /// <param name="obj"></param>
-    /// <param name="converter"></param>
-    /// <returns></returns>
-    private List<ApplicationObject> FlattenCommitObject(object obj, ISpeckleConverter converter)
+    /// <param name="obj">The root <see cref="Base"/> object to traverse</param>
+    /// <param name="converter">The converter instance, used to define what objects are convertable</param>
+    /// <returns>A flattened list of objects to be converted ToNative</returns>
+    private List<ApplicationObject> FlattenCommitObject(Base obj, ISpeckleConverter converter)
     {
-      var objects = new List<ApplicationObject>();
-
-      if (obj is Base @base)
+      void StoreObject(Base b)
       {
-        var appObj = new ApplicationObject(@base.id, ConnectorCSIUtils.SimplifySpeckleType(@base.speckle_type))
-        {
-          applicationId = @base.applicationId,
-          Status = ApplicationObject.State.Unknown
-        };
+        if (!StoredObjects.ContainsKey(b.id))
+          StoredObjects.Add(b.id, b);
+      }
 
-        if (converter.CanConvertToNative(@base))
+      ApplicationObject CreateApplicationObject(Base current)
+      {
+        ApplicationObject NewAppObj()
```

Comment:
Am I seeing this right? Is this a local function inside a local function inside a class method? 🥲 😅

Reply:
Not quite,
This is two local functions, inside a class method, side by side, no deeper nesting

Reply:
Wait!! I'm blind,
yes there does appear to be a local function within a local function 😭
More evidence that we need to extract this to a helper class.

Reply:
I trust your judgement regarding if this should be "fixed" now, or rather open a different ticket for it and merge this in. Let me know and I'll approve+merge if necessary.

Reply:
This is the same as in  rhino connector (and maybe others too). for now I think it's fine.
It sounds like we might be rewriting some of this for dui3 anyhow.